### DISCUSSION TITLE
Stuck partial wake locks rate above bad behaviour threshold

### DISCUSSION THREAD

**Discussion by ilushaaa** [airsdk/Adobe-Runtime-Support#1381] [Q&A] [answered]
Hello,

In the Google Play Console for my game I see stuck partial wake locks (background) rate 0.27% and the threshold is 0.1%. So the game gets Bad behaviour label.

What could be the reason for this error? What happens to the app after minimization? Can air in a minimized state still use some system resources? What happened with started timers?
I also use a number of extensions including adverts from distriqt. And this extension needs to be specified in a manifest the permission: 
`<uses-permission android:name="android.permission.WAKE_LOCK" />`
How necessary is this permission and can it affect the error?

Any help would be appreciated because Google stops promoting apps which have bad behaviour labels.

Thanks

**Comment by ilushaaa** [comment, marked as answer]
Hi @ajwfrost, could you at least take a peek at the problem?) I just want to know if any part of Air runtime can run any activity in the deactivated state (when an application is in the background). For example, we monitor the state of the network via flash.net.NetworkInfo. Or do we need to look for the cause in the connected ANEs? 
Thanks

Update. When switching to the background, the music theme had the volume set to 0. After changing to stop playing, the problem went away. It turns out that this is a known issue, and it has been discussed before.

**Reply by ajwfrost** [reply]
Hi

AIR doesn't use partial wakelocks in itself... no hint of `PARTIAL_WAKE_LOCK` in the code other than in a PlayPanel service (I suspect this isn't really used much...)

https://developer.android.com/topic/performance/vitals/wakelock
https://helpx.adobe.com/playpanel/release-note/playpanel-release-notes.html

But, when AIR is in the background it should be ticking over around 4fps so there may still be chances for activity to happen... timers etc all still kick off.

thanks

**Reply by ilushaaa** [reply]
Thanks for the clarification
